client/src/app/site/assignments/components/assignment-poll/assignment-poll-dialog.component.ts

```diff
@@ -0,0 +1,148 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA, MatSnackBar } from '@angular/material';
+import { TranslateService } from '@ngx-translate/core';
+
+import { AssignmentPollService } from '../../services/assignment-poll.service';
+import { CalculablePollKey } from 'app/core/ui-services/poll.service';
+import { Poll } from 'app/shared/models/assignments/poll';
+import { PollOption, PollVoteValue } from 'app/shared/models/assignments/poll-option';
+import { ViewUser } from 'app/site/users/models/view-user';
+
+/**
+ * A dialog for updating the values of an assignment-related poll.
+ */
+@Component({
+    selector: 'os-assignment-poll-dialog',
+    templateUrl: './assignment-poll-dialog.component.html',
+    styleUrls: ['./assignment-poll-dialog.component.scss']
+})
+export class AssignmentPollDialogComponent {
+    /**
+     * List of accepted special non-numerical values from {@link PollService}
+     */
+    public specialValues: [number, string][];
```

Comment:
nice name ;)
What is the exact content for `[number, string]`?

Reply:
it's the '-1: majority, -2: undocumented' stuff. There was discussion a while back about more values to be included, so I didn't want to hard code here